rt_entt_codegen/shared/parallel.cc

```diff
@@ -0,0 +1,233 @@
+	std::vector<std::vector<ecsact_system_like_id>>&   parallel_system_cluster
+) -> void;
+
+auto ecsact::rt_entt_codegen::parallel::print_parallel_system_executions(
+	ecsact::codegen_plugin_context&                     ctx,
+	const ecsact::rt_entt_codegen::ecsact_entt_details& details,
+	std::vector<ecsact_system_like_id>                  system_list,
+	bool                                                is_child
+) -> void {
+	using ecsact::cc_lang_support::cpp_identifier;
+	using ecsact::cpp_codegen_plugin_util::block;
+
+	auto parallel_system_cluster =
+		std::vector<std::vector<ecsact_system_like_id>>{};
+
+	loop_iterator(system_list, system_list.begin(), parallel_system_cluster);
+
+	for(const auto& systems_to_parallel : parallel_system_cluster) {
+		if(systems_to_parallel.size() == 1) {
+			auto sync_sys_id = systems_to_parallel[0];
+
+			auto sync_sys_name =
+				cpp_identifier(ecsact::meta::decl_full_name(sync_sys_id));
+
+			if(details.is_action(sync_sys_id)) {
+				if(is_child) {
+					ctx.write(std::format(
+						"ecsact::entt::execute_actions<{}>(registry, parent_context, "
+						"actions_map);\n",
+						sync_sys_name
+					));
+				} else {
+					ctx.write(std::format(
+						"ecsact::entt::execute_actions<{}>(registry, nullptr, "
+						"actions_map);\n",
+						sync_sys_name
+					));
+				}
+			}
+			if(details.is_system(sync_sys_id)) {
+				if(is_child) {
+					ctx.write(std::format(
+						"ecsact::entt::execute_system<{}>(registry, parent_context, "
+						"actions_map);\n",
+						sync_sys_name
+					));
+				} else {
+					ctx.write(std::format(
+						"ecsact::entt::execute_system<{}>(registry, nullptr, "
+						"actions_map);\n",
+						sync_sys_name
+					));
+				}
+			}
+			continue;
+		}
+
+		if(is_child) {
+			ctx.write("ecsact_execute_parallel_cluster(registry, parent_context, ");
+		} else {
+			ctx.write("ecsact_execute_parallel_cluster(registry, nullptr, ");
+		}
+		ctx.write("std::array {\n");
+		for(const auto system_like_id : systems_to_parallel) {
+			auto cpp_decl_name =
+				cpp_identifier(ecsact::meta::decl_full_name(system_like_id));
+
+			if(details.is_action(system_like_id)) {
+				ctx.write(
+					"std::pair{&ecsact::entt::execute_actions<",
+					cpp_decl_name,
+					">, actions_map},\n"
+				);
+			} else if(details.is_system(system_like_id)) {
+				ctx.write(
+					"std::pair{&ecsact::entt::execute_system<",
+					cpp_decl_name,
+					">, actions_map},\n"
+				);
+			} else {
+				ctx.write("// ??? unhandled ??? ", cpp_decl_name, "\n");
+			}
+		}
+		ctx.write("});\n");
+	}
+}
+
+auto is_capability_safe(ecsact_system_capability capability) -> bool {
+	if(capability == ECSACT_SYS_CAP_ADDS ||
+		 capability == ECSACT_SYS_CAP_REMOVES ||
+		 capability == ECSACT_SYS_CAP_READWRITE ||
+		 capability == ECSACT_SYS_CAP_WRITEONLY ||
+		 capability == ECSACT_SYS_CAP_OPTIONAL_WRITEONLY ||
+		 capability == ECSACT_SYS_CAP_OPTIONAL_READWRITE) {
+		return false;
+	}
```

Comment:
Since the capabilities are bitflags we could simply check for the adds, removes, and write bits. Just a thought - I wouldn't reject your PR for this.